rpc/utils.go

```diff
@@ -250,5 +251,13 @@ func NewID() ID {
 			val >>= 8
 		}
 	}
-	return ID("0x" + hex.EncodeToString(id))
+
+	rpcId := hex.EncodeToString(id)
+	// rpc ID's are RPC quantities, no leading zero's and 0 is 0x0
+	rpcId = strings.TrimLeft(rpcId, "0")
+	if rpcId == "" {
+		rpcId = "0"
+	}
+
+	return ID("0x" + rpcId)
```

Comment:
Fun fact. HexNumer.MarshalJSON could re reduced to that last line in my suggestion above.